employee-rostering-webapp/src/test/java/org/optaweb/employeerostering/webapp/shift/ShiftRestServiceIT.java

```diff
@@ -0,0 +1,189 @@
+        assertClientResponseOk();
+        return out;
+    }
+
+    @After
+    public void cleanup() {
+        deleteTestTenant();
+    }
+
+    @Test
+    public void testDeleteNonExistingShift() {
+        createTestTenant();
+        final long nonExistingShiftId = 123456L;
+        boolean result = shiftRestService.removeShift(TENANT_ID, nonExistingShiftId);
+        assertThat(result).isFalse();
+        assertClientResponseOk();
+    }
+
+    @Test
+    public void testUpdateNonExistingShift() {
+        createTestTenant();
+        final long nonExistingShiftId = 123456L;
+        Spot spot = createSpot("spot");
+        Employee employee = createEmployee("employee");
+        Employee rotationEmployee = createEmployee("rotationEmployee");
+        LocalDateTime startDateTime = LocalDateTime.of(2000, 1, 1, 0, 0, 0, 0);
+        LocalDateTime endDateTime = startDateTime.plusHours(8);
+
+        ShiftView nonExistingShift = new ShiftView(TENANT_ID, spot, startDateTime, endDateTime);
+        nonExistingShift.setEmployeeId(employee.getId());
+        nonExistingShift.setRotationEmployeeId(rotationEmployee.getId());
+        nonExistingShift.setId(nonExistingShiftId);
+        ShiftView updatedShift = shiftRestService.updateShift(TENANT_ID, nonExistingShift);
+        assertClientResponseOk();
+
+        assertThat(updatedShift.getSpotId()).isEqualTo(updatedShift.getSpotId());
+        assertThat(updatedShift.getEmployeeId()).isEqualTo(updatedShift.getEmployeeId());
+        assertThat(updatedShift.getStartDateTime()).isEqualTo(updatedShift.getStartDateTime());
+        assertThat(updatedShift.getEndDateTime()).isEqualTo(updatedShift.getEndDateTime());
+        assertThat(updatedShift.getRotationEmployeeId()).isEqualTo(updatedShift.getRotationEmployeeId());
+        assertThat(updatedShift.getId()).isNotNull().isNotEqualTo("spot");
+    }
+
+    @Test
+    public void testGetOfNonExistingShift() {
+        createTestTenant();
+        final long nonExistingShiftId = 123456L;
+        assertThatExceptionOfType(javax.ws.rs.NotFoundException.class)
+                .isThrownBy(() -> shiftRestService.getShift(TENANT_ID, nonExistingShiftId));
+        assertClientResponseError(Response.Status.NOT_FOUND);
+    }
+
+    @Test
+    public void testCrudShift() {
+        createTestTenant();
+        Spot spot = createSpot("spot");
+        Employee employee = createEmployee("employee");
+        Employee rotationEmployee = createEmployee("rotationEmployee");
+        LocalDateTime startDateTime = LocalDateTime.of(2000, 1, 1, 0, 0, 0, 0);
+        LocalDateTime endDateTime = startDateTime.plusHours(8);
+
+        ShiftView testAddShift = new ShiftView(TENANT_ID, spot, startDateTime, endDateTime);
+        shiftRestService.addShift(TENANT_ID, testAddShift);
+        assertClientResponseOk();
+
+        List<ShiftView> shifts = shiftRestService.getShifts(TENANT_ID);
+        assertClientResponseOk();
+        assertThat(shifts)
```

Comment:
The assertion is incomplete - you just say "how", not "what".

Reply:
Fixed; verified the shifts list contains testAddShift and only testAddShift